### PR TITLE
RoboCup Changes: CreaseDefenderTactic whitelist and tests

### LINKED ISSUE #901: RoboCup Changes: Port CreaseDefenderTactic
### Description of the task

Cleanup and migrate the changes made to the CreaseDefenderTactic at RoboCup 2019 to the current master branch.

### Acceptance criteria

- [ ] Migrate changes

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Minor whitelist change from RoboCup 2019

### Testing Done

Added a few unit tests. By no means are these a complete test suite but they are something to verify very basic behavior. Tactics will be fully tested once all RoboCup changes are merged.

### Resolved Issues

resolves #901 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/hl/stp/tactic/crease_defender_tactic_test.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    // Check an intent was returned (the pointer is not null)
+    EXPECT_TRUE(intent_ptr);
+
+    try
+    {
+        MoveIntent move_intent = dynamic_cast<MoveIntent &>(*intent_ptr);
+        EXPECT_TRUE(move_intent.getDestination().isClose(
+            Point(world.field().friendlyDefenseArea().posXPosYCorner().x() +
+                      ROBOT_MAX_RADIUS_METERS,
+                  0.18),
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/crease_defender_tactic_test.cpp]: Can you comment where this .18 comes from?

File: src/software/ai/hl/stp/tactic/crease_defender_tactic_test.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    // Check an intent was returned (the pointer is not null)
+    EXPECT_TRUE(intent_ptr);
+
+    try
+    {
+        MoveIntent move_intent = dynamic_cast<MoveIntent &>(*intent_ptr);
+        EXPECT_TRUE(move_intent.getDestination().isClose(
+            Point(world.field().friendlyDefenseArea().posXPosYCorner().x() +
+                      ROBOT_MAX_RADIUS_METERS,
+                  -0.18),
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/crease_defender_tactic_test.cpp]: and this -.18?

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Looks good, just a couple of confusing constants. It looks like .18 is robot diameter. Can you use 2*ROBOT_MAX_RADIUS_METERS instead?

File: src/software/ai/hl/stp/tactic/crease_defender_tactic_test.cpp (hunk 2 above)
Comment (MathewMacDougall) [reply src/software/ai/hl/stp/tactic/crease_defender_tactic_test.cpp]: added comment

File: src/software/ai/hl/stp/tactic/crease_defender_tactic_test.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/software/ai/hl/stp/tactic/crease_defender_tactic_test.cpp]: added comment

Comment (jonathanlew) [reply src/software/ai/hl/stp/tactic/crease_defender_tactic_test.cpp]: Can we replace .18 with 2*ROBOT_MAX_RADIUS_METERS?

Comment (MathewMacDougall) [reply src/software/ai/hl/stp/tactic/crease_defender_tactic_test.cpp]: done

**Review by jonathanlew — APPROVED** [review]
Looks good! (sorry for being so picky)

**Review by garethellis0 — APPROVED** [review]

**Pull request merged**